bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationConfigurationHandler.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.internal.kostal.inverter.secondgeneration;
+
+import java.security.MessageDigest;
+import java.util.Base64;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link SecondGenerationConfigurationHandler} is responsible for configuration changes,
+ * regarded to second generation part of the binding.
+ *
+ * @author Örjan Backsell - Initial contribution Piko1020, Piko New Generation
+ */
+@NonNullByDefault
+public class SecondGenerationConfigurationHandler {
+
+    private static Logger logger = LoggerFactory.getLogger(SecondGenerationConfigurationHandler.class);
+
+    public static void executeConfigurationChanges(HttpClient httpClient, String url, String username, String password,
+            String dxsId, String value) throws Exception {
+        httpClient.start();
+        String urlLogin = url + "/api/login.json?";
+        String salt = "";
+        String sessionId = "";
+
+        String getAuthenticateResponse = httpClient.GET(urlLogin).getContentAsString();
+        try {
+            JsonObject getAuthenticateResponseJsonObject = (JsonObject) new JsonParser()
+                    .parse(transformJsonResponse(getAuthenticateResponse));
+
+            sessionId = extractSessionId(getAuthenticateResponseJsonObject);
+
+            JsonObject authenticateJsonObject = new JsonParser().parse(getAuthenticateResponse.toString())
+                    .getAsJsonObject();
+            salt = authenticateJsonObject.get("salt").getAsString();
+
+            String saltedPassword = new StringBuilder(password).append(salt).toString();
+            MessageDigest mDigest = MessageDigest.getInstance("SHA1");
+
+            byte[] mDigestedPassword = mDigest.digest(saltedPassword.getBytes());
```

Comment:
Can you specify the charset you expect? E.g.
```suggestion
            byte[] mDigestedPassword = mDigest.digest(saltedPassword.getBytes(StandardCharsets.UTF_8));
```